src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/NotificationService.java

```diff
@@ -0,0 +1,129 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.niis.xroad.restapi.service;
+
+import ee.ria.xroad.common.CodedException;
+import ee.ria.xroad.commonui.SignerProxy;
+import ee.ria.xroad.signer.protocol.dto.TokenInfo;
+
+import lombok.extern.slf4j.Slf4j;
+import org.niis.xroad.restapi.dto.AlertStatus;
+import org.niis.xroad.restapi.facade.GlobalConfFacade;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.Optional;
+
+/**
+ * service class for handling notifications
+ */
+@Slf4j
+@Service
+@Transactional
+@PreAuthorize("isAuthenticated()")
+public class NotificationService {
+    private static OffsetDateTime backupRestoreRunningSince;
+    private final GlobalConfFacade globalConfFacade;
+    private final TokenService tokenService;
+
+    /**
+     * constructor
+     */
+    @Autowired
+    public NotificationService(GlobalConfFacade globalConfFacade, TokenService tokenService) {
+        this.globalConfFacade = globalConfFacade;
+        this.tokenService = tokenService;
+    }
+
+    /**
+     * Checks the status of system alerts that may affect whether the system
+     * is operational or not. If alerts are disabled, the status of all alerts true.
+     * @return
+     */
+    public AlertStatus getAlerts() {
+        AlertStatus alertStatus = new AlertStatus();
+        OffsetDateTime backupRestoreStartedAt = getBackupRestoreRunningSince();
+        if (backupRestoreStartedAt != null) {
+            alertStatus.setBackupRestoreRunningSince(backupRestoreStartedAt);
+            alertStatus.setCurrentTime(OffsetDateTime.now(ZoneOffset.UTC));
+        }
+        alertStatus.setGlobalConfValid(isGlobalConfValid());
+        alertStatus.setSoftTokenPinEntered(isSoftTokenPinEntered());
+        return alertStatus;
+    }
+
+    /**
+     * Verifies that the global configuration is valid.
+     * @return
+     */
+    private boolean isGlobalConfValid() {
+        try {
+            globalConfFacade.verifyValidity();
+            return true;
+        } catch (CodedException e) {
+            return false;
+        }
+    }
+
+    /**
+     * Checks if soft token pin is entered.
+     * @return
+     */
+    private boolean isSoftTokenPinEntered() {
+        Optional<TokenInfo> token = tokenService.getAllTokens().stream()
+                .filter(t -> t.getId().equals(SignerProxy.SSL_TOKEN_ID)).findFirst();
+        if (!token.isPresent()) {
+            throw new RuntimeException("soft token not found");
```

Comment:
The old implementation does not stop execution if the soft token is not found. However a RuntimeException here makes sense if the token has been properly initialized but nevertheless is not found. Adding the init check in the `getAlerts` method would actually help in this situation too

Reply:
Yes, adding the init check in the getAlerts method will solve this too. I'll create a new ticket about that.